### ISSUE TITLE
Advice: Make this adapter as a homekit manager.

### ISSUE THREAD

**Issue by SchumyHao** [ioBroker/ioBroker.ham#22]
Hi @Apollon77 

This adapter is an brilliant adapter for ioBroker:

- It can translate all devices that supported by homekit to ioBroker. 
- User can add `homebridge-plugin-*` in frontend, without open SSH terminal and tun npm command.(It is not a easy thing for new user)
- User can edit plugin JSON config file in frontend, and frontend can check JSON format.(If second thing is not easy for new user, *EDIT JSON FILE* is a nightmare for them!)

As this adapter is based on [homebridge-plugin-wrapper](https://github.com/Apollon77/homebridge-plugin-wrapper), this lib's README file says it can not work as a normal bridge, user can not find this bridge in Homekit. It's a pity that this adapter can not work as a normal homekit bridge.

Hi @Apollon77 , can you give me a hit that why we can not make adapter into a homekit magager?
If default mode can work as a normal homekit bridge, we can achive all benefits together:
- Install HAP with this adapter, no need shell command for user.
- Install plugins in frontend.
- Edit plugins in frontend.
- Help user check his config file.
- A *FULL FUNCTION* homekit bridge. Homekit function works normal
- User can use ioBroker's blockly and JS to edit automations.
- User can make there own website for his house by UI adapters supported by ioBroker.

I think if this adapter can attracts HAP user to use ioBroker.

BRs
Schumy

**Comment by Apollon77** [comment]
hey,

the adapter supports two modes: First mode is using the "homebridgeplugin-wrapper" and acting as a plugin wrapper only. econd way is the so called "global mode" where you can have your own homebridge installation (which is also seen as bridge and so reachable) and the adapter is also using this instance - then the config comes from your "global homebridge" rather then from the wrapped one but all data are available.

Maybe the gloabl mode is what you want?

**Comment by SchumyHao** [comment]
> hey,
> 
> the adapter supports two modes: First mode is using the "homebridgeplugin-wrapper" and acting as a plugin wrapper only. econd way is the so called "global mode" where you can have your own homebridge installation (which is also seen as bridge and so reachable) and the adapter is also using this instance - then the config comes from your "global homebridge" rather then from the wrapped one but all data are available.
> 
> Maybe the gloabl mode is what you want?

Hi @Apollon77 , thank you for your reminder.

But I think globle mode is not the best way:
- In globle mode, user have to install homebridge in shell console by npm command
- User have to install plugins by npm command
- User heve to Edit plugins in console, without JSON check enabled.

I know those 3 points is not so necessary for advice user, but I found there are lots of new users make mistakes, they want to accomplish what they can in frontend, not in console.
They think console like coding, user experence is not good.

As wrapper mode has already done the development work for those 3 points, can we just port them to globle mode?

**Comment by Mr-REM-Esq** [comment]
Perfect idea! It is really very difficult to edit the JSON-file when you don't know what is it and how it should look.

**Comment by kirovilya** [comment]
HAM need to be cooperate with [Yahka](https://github.com/jensweigele/ioBroker.yahka) :)

**Comment by Apollon77** [comment]
@SchumyHao Aaaahhh so the idea would be a third mode like "wrapper but published as Homebridge instance" where everything is like wrapper but publishing the bridge in the network.

The main topic why this was not done are the many dependencies I would need to add amd especially installing the "curve" encryption stuff always makes often problems on install and I wanted to get rid of them. Additionally you need MDNS support and and and (see Yahka about the list of additionally needed stuff - and if not there it will not build)

So to have full "publishing" support I also need all the encryption things with all the other dependencies. And so it would be way more stuff and problems. Ok could be defined as "optionalDep" in package, so that it is not problematic when failed.

Need to think because then it there is no "convienenmt way for the user" ... maybe as a second separate adapter (ham-homebridge) where only this mode works and only this adapter will have higher "needs to be done before installing" ... What do you think about this?

@Mr-REM-Esq To come around the fact to edit the json would be a big topic because each plugin is different

@kirovilya In fact HAM is "using homebridge data in ioBroker" and Yahka is the other way, so more "publishig iobroker data to Homebridge". If yahka would be a homebridge plugin then it could be used "out of the box" and only need to care about the data handling

**Comment by SchumyHao** [comment]
OK Thank you for your detail explaination.
I'll think about this issue, how to handle the dependencies.
Thank you for your exellent adapter~

**Comment by Apollon77** [comment]
Lets leave it open for referece, discussion and thinking

**Comment by yaming116** [comment]
hey,

i add an optional option to edit profile information using the web editor in global mode #23

**Comment by SchumyHao** [comment]
> hey,
> 
> @SchumyHao @Apollon77
> 
> i add an optional option to edit profile information using the web editor in global mode #23

It seems can edit global config json file with JSON check in frontend.

**Comment by Apollon77** [comment]
The upcoming 1.1.0 adds a new "Local" mode. When selected it will install am"local" homebridge and start this like a global one. Config is edited by the adapter Admin UI. You meaned something like this?

**Comment by Apollon77** [comment]
1.1.1
